JavaScript/chapter01/1.8 - Zero Matrix/rroque98_sol.js

```diff
@@ -0,0 +1,55 @@
+/*Write an algorithm such that if an element
+ in an M*N matrix is 0,it's entire row and column
+  are set to 0*/
+
+const zeroMatrix = nestedArr => {
+  const rowsAndColsWithZeros = checkForZeroIndex(nestedArr);
+  const rowLength = nestedArr[0].length;
+  const columnLength = nestedArr.length;
+  for (const rowIndex of rowsAndColsWithZeros['rows']) {
+    for (let x = 0; x < rowLength; x++) {
+      nestedArr[rowIndex][x] = 0;
+    }
+  }
+  for (const colIndex of rowsAndColsWithZeros['columns']) {
+    for (let x = 0; x < columnLength; x++) {
+      nestedArr[x][colIndex] = 0;
+    }
+  }
+  return nestedArr;
+
+  // **** Helper functions ****
+  function checkForZeroIndex(nestArr) {
```

Comment:
If you make this a free function instead of nesting it within this other function, you can test it independently.

Reply:
I made the helper function free now, and I included some tests for the helper function